src/lib/swapper/swappers/LifiSwapper/LifiSwapper2.ts

```diff
@@ -0,0 +1,125 @@
+import type { ChainKey, GetStatusRequest, Route } from '@lifi/sdk/dist/types'
+import type { AssetId, ChainId } from '@shapeshiftoss/caip'
+import type { ChainAdapter, EvmChainId } from '@shapeshiftoss/chain-adapters'
+import type { ETHSignTx } from '@shapeshiftoss/hdwallet-core'
+import type { Result } from '@sniptt/monads/build'
+import { Err } from '@sniptt/monads/build'
+import { getChainAdapterManager } from 'context/PluginProvider/chainAdapterSingleton'
+import type { Asset } from 'lib/asset-service'
+import type {
+  BuyAssetBySellIdInput,
+  ExecuteTradeArgs,
+  GetEvmTradeQuoteInput,
+  GetTradeQuoteInput,
+  GetUnsignedTxArgs,
+  SwapErrorRight,
+  Swapper2,
+  TradeQuote2,
+} from 'lib/swapper/api'
+import { getLifi } from 'lib/swapper/swappers/LifiSwapper/utils/getLifi'
+
+import { filterEvmAssetIdsBySellable } from '../utils/filterAssetIdsBySellable/filterAssetIdsBySellable'
+import { filterCrossChainEvmBuyAssetsBySellAssetId } from '../utils/filterBuyAssetsBySellAssetId/filterBuyAssetsBySellAssetId'
+import { getTradeQuote } from './getTradeQuote/getTradeQuote'
+import { getLifiChainMap } from './utils/getLifiChainMap'
+import { getUnsignedTx } from './utils/getUnsignedTx/getUnsignedTx'
+
+const executedTrades: Map<string, GetStatusRequest> = new Map()
+const tradeQuoteMetadata: Map<string, Route> = new Map()
+
+let lifiChainMapPromise: Promise<Result<Map<ChainId, ChainKey>, SwapErrorRight>> | undefined
+
+export const lifi: Swapper2 = {
+  getTradeQuote: async (
+    input: GetTradeQuoteInput,
+    assets: Partial<Record<AssetId, Asset>>,
+    sellAssetPriceUsdPrecision: string,
+  ): Promise<Result<TradeQuote2, SwapErrorRight>> => {
+    if (lifiChainMapPromise === undefined) lifiChainMapPromise = getLifiChainMap()
+
+    const maybeLifiChainMap = await lifiChainMapPromise
+
+    if (maybeLifiChainMap.isErr()) return Err(maybeLifiChainMap.unwrapErr())
+
+    const tradeQuoteResult = await getTradeQuote(
+      input as GetEvmTradeQuoteInput,
+      maybeLifiChainMap.unwrap(),
+      assets,
+      sellAssetPriceUsdPrecision,
+    )
+
+    return tradeQuoteResult.map(({ selectedLifiRoute, ...tradeQuote }) => {
+      const { receiveAddress, affiliateBps } = input
+      const id = selectedLifiRoute.id
+
+      // store the lifi quote metadata for transaction building later
+      tradeQuoteMetadata.set(id, selectedLifiRoute)
+
+      return { id, receiveAddress, affiliateBps, ...tradeQuote }
+    })
+  },
+
+  getUnsignedTx: async ({ from, tradeQuote, stepIndex }: GetUnsignedTxArgs): Promise<ETHSignTx> => {
```

Comment:
note lack of wallet input requirement :D

Reply:
Lovely - pretty straightforward for EVM chains, all we need is a `from`!